### PR TITLE
Update ld_default package with non-interactive mode support

### PR DESCRIPTION
**Pull request by JL2210**
Adds ability for ld_default to be ran in non-interactive mode (will be useful for packages).
Uses double square brackets, a more modern Bash-only feature that can evaluate strings with spaces in them. See [this link](https://serverfault.com/questions/52034/what-is-the-difference-between-double-and-single-square-brackets-in-bash) for more details.
Provides a fallback if linker type is unknown instead of spewing random binary data.
Adds message to postinstall about aforementioned added features.

### CODE REVIEW

**Comment by uberhacker** [comment]
This is a good idea but you overlooked the possibility that you might need to return to the previous linker default after the package process is complete.  You would need a way to assign the current default, do your package processing with the new default and then return to what it was previously.

**Comment by JL2210** [comment]
How would I go about doing that?

**Comment by uberhacker** [comment]
I think you misunderstood what I was trying to communicate.  In the package that needs the new default linker, you would need to doing something like this:
```
@old_default = `ld_default d`
@new_default = `ld_default g`
...
system "ld_default #{@old_default}"
```
The `d` parameter is what grabs the current default linker.

**Comment by JL2210** [comment]
Yes. See this package: 
```
class V4l_utils < Package
  description 'A conglomeration of video4linux libraries and utilities.'
  homepage 'https://linuxtv.org/'
  version '1.16.3'
  source_url 'https://linuxtv.org/downloads/v4l-utils/v4l-utils-1.16.3.tar.bz2'
  source_sha256 '7c5c0d49c130cf65d384f28e9f3a53c5f7d17bf18740c48c40810e0fbbed5b54'

  binary_url ({
  })
  binary_sha256 ({
  })

  #depends_on 'sdl2_image'
  depends_on 'libsdl2'
  depends_on 'libglu'
  depends_on 'alsa_lib'
  depends_on 'ld_default'

  def self.build
    original_default = "`ld_default g`"
    system "./configure",
           "--disable-static",
           "--prefix=#{CREW_PREFIX}",
           "--libdir=#{CREW_LIB_PREFIX}"
    system "ld_default #{original_default}"
    system 'make'
  end

  def self.install
    system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
  end
end
```

File: packages/ld_default.rb [hunk 1]
```diff
@@ -21,28 +21,42 @@ def self.build
...
+    echo
+    echo '  Enter the new default linker:'
+    echo
+    echo '  b = ld.bfd'
+    echo '  g = ld.gold'
+    echo '  l = ld.lld'
+    echo '  0 = Cancel'
+    echo
+  else
+    echo \"\${current}\" | cut -d'.' -f2 | cut -c1
```
Comment (uberhacker) [inline packages/ld_default.rb]: This could be shortened to `echo \"\${current}\" | cut -c1`.

Comment (uberhacker) [reply packages/ld_default.rb]: Nevermind.  This is correct.

File: packages/ld_default.rb [hunk 2]
```diff
@@ -21,28 +21,42 @@ def self.build
...
+  fi
 
   while true; do
-    read default
-    case \${default} in
+    if [[ -z \"\${1}\" ]]; then
+      read default
+    else
+      default=\"\${1}\"
+      if [[ \"\${default}\" != 'b' ]] && [[ \"\${default}\" != 'g' ]] && [[ \"\${default}\" != 'l' ]]; then
```
Comment (uberhacker) [inline packages/ld_default.rb]: See if this works:
```
if ! [[ \"\${default}\" =~ "^[b|g|l]$" ]]; then

Comment (JL2210) [reply packages/ld_default.rb]: No. It doesn't.
```
chronos@localhost ~/Download/source/chromebrew $ ld_default b
l
Invalid linker configuration: b
chronos@localhost ~/Download/source/chromebrew $ ld_default g
l
Invalid linker configuration: g
chronos@localhost ~/Download/source/chromebrew $ ld_default l
```

Comment (uberhacker) [reply packages/ld_default.rb]: You might need to escape more characters.  I'm thinking `|` is one of them.

Comment (uberhacker) [reply packages/ld_default.rb]: Anyway, not worth wrestling with it unless you want to do it.

Comment (JL2210) [reply packages/ld_default.rb]: I don't think `|` is a special character in Ruby. I'd have to check, though.

**Pull request merged**